### PR TITLE
provider/azurerm: EventHub Authorization Rules

### PR DESCRIPTION
**Pull request by tombuildsstuff**
- [x] CRUD
- [x] Import
- [x] Documentation
- [x] Tests
  - [x] Acceptance Tests
  - [x] Import Tests
  - [x] Results posted in the Comments

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
One important thing to note here:

Access Rights have three possible values: `Listen`, `Send` and `Manage`. The SDK exposes this as an array of strings - however there's some validation around these values, in particular:
 - A minimum of one value needs to be sent
 - Each value can only be sent a single time
 - When `Manage` is enabled - `Listen` and `Send` need to be present

For this reason I've modelled this as three separate bool's rather than an array of strings. I attempted to make AccessRights/Permissions a sub-type - but as you can't apply validation against a Set it's not possible to handle this validation at that level (and thus it loses any benefit); thus instead I've documented the limitation and validated at least one is selected in the CreateUpdate call.

That's not ideal - but I'm unsure how else to do that for now?

**Comment by tombuildsstuff** [comment]
Tests pass:

```
$ TF_ACC=1 go test ./builtin/providers/azurerm -v -run TestAccAzureRMEventHubAuthorization -timeout 120m
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importListen
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importListen (365.31s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importSend
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importSend (360.65s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importReadWrite
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importReadWrite (362.65s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importManage
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importManage (359.12s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_listen
--- PASS: TestAccAzureRMEventHubAuthorizationRule_listen (364.67s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_send
--- PASS: TestAccAzureRMEventHubAuthorizationRule_send (363.26s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_readwrite
--- PASS: TestAccAzureRMEventHubAuthorizationRule_readwrite (365.47s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_manage
--- PASS: TestAccAzureRMEventHubAuthorizationRule_manage (364.11s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	2905.273s
```

File: builtin/providers/azurerm/resource_arm_eventhub_authorization_rule.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	log.Printf("[INFO] preparing arguments for Azure ARM EventHub Authorization Rule creation.")
+
+	name := d.Get("name").(string)
+	namespaceName := d.Get("namespace_name").(string)
+	eventHubName := d.Get("eventhub_name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	rights := expandEventHubAuthorizationRuleAccessRights(d)
+	if len(*rights) == 0 {
```
Comment (stack72) [inline builtin/providers/azurerm/resource_arm_eventhub_authorization_rule.go]: is this the best way to do this?

**Comment by pmcatominey** [comment]
@tombuildsstuff you could check that the correct values are present after `expand`ing, returning an error in the create function before making the network call will have the same result as a validation check failing.

**Comment by tombuildsstuff** [comment]
Updated as suggested - I've also included some validation that when Manage is enabled Listen/Send must be too.

Tests pass:

```
$ make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMEventHubAutho'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/02 14:44:45 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/azurerm -v -run=TestAccAzureRMEventHubAutho -timeout 120m
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importListen
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importListen (396.68s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importSend
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importSend (366.62s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importReadWrite
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importReadWrite (366.15s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importManage
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importManage (366.29s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_listen
--- PASS: TestAccAzureRMEventHubAuthorizationRule_listen (365.01s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_send
--- PASS: TestAccAzureRMEventHubAuthorizationRule_send (368.70s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_readwrite
--- PASS: TestAccAzureRMEventHubAuthorizationRule_readwrite (360.24s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_manage
--- PASS: TestAccAzureRMEventHubAuthorizationRule_manage (360.71s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	2950.420s
```

**Comment by stack72** [comment]
Thanks @tombuildsstuff :) This LGTM now

```
% make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMEventHubAuthorization'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/02 15:40:34 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/azurerm -v -run=TestAccAzureRMEventHubAuthorization -timeout 120m
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importListen
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importListen (380.63s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importSend
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importSend (368.15s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importReadWrite
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importReadWrite (367.66s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_importManage
--- PASS: TestAccAzureRMEventHubAuthorizationRule_importManage (376.62s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_listen
--- PASS: TestAccAzureRMEventHubAuthorizationRule_listen (366.65s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_send
--- PASS: TestAccAzureRMEventHubAuthorizationRule_send (366.23s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_readwrite
--- PASS: TestAccAzureRMEventHubAuthorizationRule_readwrite (366.85s)
=== RUN   TestAccAzureRMEventHubAuthorizationRule_manage
--- PASS: TestAccAzureRMEventHubAuthorizationRule_manage (373.25s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	2966.060s
```

**Pull request merged**